lib/public_suffix.ex

```diff
@@ -55,13 +54,74 @@ defmodule PublicSuffix do
       iex> registrable_domain("foo.github.io")
       "foo.github.io"
   """
-  @spec registrable_domain(String.t) :: nil | String.t
   @spec registrable_domain(String.t, options) :: nil | String.t
   def registrable_domain(domain, options \\ []) when is_binary(domain) do
     # "The registered or registrable domain is the public suffix plus one additional label."
     parse_domain(domain, options, 1)
   end
 
+  @doc """
+  Parses the provided domain and returns the prevailing rule based on the
+  publicsuffix.org rules. If no rules match, the prevailing rule is "*",
+  unless the provided domain has a leading dot, in which case the input is
+  invalid and the function returns `nil`.
+
+  ## Examples
+
+      iex> prevailing_rule("foo.bar.com")
+      "com"
+      iex> prevailing_rule("co.uk")
+      "co.uk"
+      iex> prevailing_rule("foo.ck")
+      "*.ck"
+      iex> prevailing_rule("foobar.example")
+      "*"
+
+  You can use the `ignore_private` keyword to exclude private (non-ICANN) domains.
+
+      iex> prevailing_rule("foo.github.io", ignore_private: false)
+      "github.io"
+      iex> prevailing_rule("foo.github.io", ignore_private: true)
+      "io"
+      iex> prevailing_rule("foo.github.io")
+      "github.io"
+  """
+  @spec prevailing_rule(String.t, options) :: nil | String.t
+  def prevailing_rule(domain, options \\ [])
+  def prevailing_rule("." <> _domain, _), do: nil
+  def prevailing_rule(domain, options) when is_binary(domain) do
+    domain
+    |> String.downcase
+    |> String.split(".")
+    |> find_prevailing_rule(options)
+    |> case do
+         {:exception, rule} -> "!" <> Enum.join(rule, ".")
+         {_, rule} -> Enum.join(rule, ".")
```

Comment:
I think I'd prefer matching on `{:normal, rule}` here (and in the similar case statement below) so as to be more explicit and assertive (which is one of the [values of elixir](http://blog.plataformatec.com.br/2014/09/writing-assertive-code-with-elixir/)).